### PR TITLE
[Components with more than 1 service] LPS-175886 Refactor DTO Converter components with more than 1 service type from Analytics Cloud

### PR DESCRIPTION
**Pull request by MrJohn1911**

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud

**Comment by catarineaguiar** [comment]
ci:test:analytics-cloud

**Comment by catarineaguiar** [comment]
no regressions found.

File: modules/apps/analytics/analytics-settings-rest-impl/src/main/java/com/liferay/analytics/settings/rest/internal/resource/v1_0/SiteResourceImpl.java [hunk 1]
```diff
@@ -116,7 +116,9 @@ private LinkedHashMap<String, Object> _getParams() {
 	@Reference
 	private Portal _portal;
 
-	@Reference
-	private SiteDTOConverter _siteDTOConverter;
+	@Reference(
+		target = "(dto.class.name=com.liferay.portal.kernel.model.Group)"
+	)
+	private DTOConverter<Group, Site> _siteDTOConverter;
```
Comment (tinatian) [inline modules/apps/analytics/analytics-settings-rest-impl/src/main/java/com/liferay/analytics/settings/rest/internal/resource/v1_0/SiteResourceImpl.java:122]: @MrJohn1911 There are two DTOConverter registered with "dto.class.name=com.liferay.portal.kernel.model.Group", so this is wrong.

**Comment by tinatian** [comment]
@MrJohn1911 As "dto.class.name" is not unique, you may have more than one impl with same property, so you can not do this.

Let's use "component.name", but I need the component be in the same module as its usages. For example, you need to move DXPEntityDTOConverterImpl to analytics-batch-export-import-impl as all its usages are there.

Please resend in small pieces.

Thanks.
Tina.

**Comment by tinatian** [comment]
@vicnate5 @catarineaguiar Please see my comments above, the code has something wrong, while the test passed, not sure if we did not have test covered or we did not trigger enough tests. Please note, the wrong code will cause random failure. Could you please check?

Thanks.
Tina.

**Comment by vicnate5** [comment]
133 of the AC tests failed here (146 on the other run), but I only see 2 tests that look unique on this PR.

https://testray.liferay.com/home/-/testray/case_results/2199801274
https://testray.liferay.com/home/-/testray/case_results/2199801715

@christianstokes Would you team be able to look at these?

See comments above, there is a code error, but we want to confirm if we missed this in tests or if we don't currently have the coverage for this.

**Comment by joshitagaki** [comment]
ci:test:analytics-cloud-acceptance

**Pull request closed without merging**